web/server/routers/account.ts

```diff
@@ -143,4 +144,40 @@ export const accountRouter = router({
         }
       }
     }),
+  get: protectedProcedure
+    .input(
+      z.object({
+        email: z.string().email(),
+        uid: z.string(),
+        name: z.string(),
+      })
+    )
+    .output(
+      z.object({
+        found: z.boolean(),
+      })
+    )
+    .query(async ({ ctx, input }) => {
+      const email = input.email;
+      if (email === null) {
+        throw new TRPCError({
+          code: "UNAUTHORIZED",
+          message: "Not authenticated",
+        });
+      } else {
+        const account = await findAccount(email);
+        if (account === null) {
+          return { found: false };
+        }
+        const user = await findUserByUid(input.uid);
+        if (!user) {
+          await createUser({
+            ...input,
+            role: account.role,
+            disabled: false,
+          });
+        }
+        return { found: true };
```

Comment:
`account` router should only modify/query the Account collection. Write another API route for User creation and have the frontend query 2 routes.